src/features/image-map-editor/image-dropzone.tsx

```diff
@@ -0,0 +1,55 @@
+import {
+  useEffect,
+  useCallback,
+} from 'react';
+import { useDropzone } from 'react-dropzone';
+import { useEditor } from './editor.context';
+import * as Styled from './image-dropzone.style';
+
+const dropzoneOptions = {
+  multiple: false,
+  accept: {'image/*': ['.png', '.jpeg', '.jpg', '.svg']}
+};
+
+const ImageDropzone = () => {
+  const {
+    acceptedFiles,
+    getRootProps,
+    getInputProps,
+  } = useDropzone(dropzoneOptions);
+  const { dispatch } = useEditor();
+
+  const addImg = useCallback((file) => {
+    const img = new Image();
+    img.src = URL.createObjectURL(file);
+    img.src = URL.createObjectURL(file);
+    img.onload = () => {
+      const imgData = {
+        el: img,
+        width: img.width,
+        height: img.height,
+        aspectRatio: img.height / img.width,
+      };
+      dispatch({ type: 'add-image', imgData })
+    };
+  }, [dispatch]);
+
+  useEffect(() => {
+    if (!acceptedFiles.length) return;
+    const [file] = acceptedFiles;
+    if (!file) return;
+    addImg(file);
+  }, [acceptedFiles, addImg]);
```

Comment:
In the react-dropzone's documents, they use `onDrop` as the useDropzone's parameter to handle files. I'd like to know why you handle files in the effect function.

Reply:
I didn't notice the `onDropAccepted` callback option. I will refactor it.